### PR TITLE
feat(infra.ci.jenkins.io) add secondary Azure VM subscription resources

### PR DESCRIPTION
**Pull request by dduportal**
Ref. https://github.com/jenkins-infra/helpdesk/issues/3818#issuecomment-1860472394

Expecting 19 new resources to be created (same as previous PRs for other controllers)

### CODE REVIEW

**Review by lemeurherve — APPROVED** [review]
> Plan: 19 to add, 0 to change, 0 to destroy.

File: infra.ci.jenkins.io.tf [hunk 1]
```diff
@@ -67,3 +67,69 @@ resource "azurerm_role_assignment" "infra_ci_jenkins_io_privatek8s_subnet_privat
...
+  provider = azurerm.jenkins-sponsorship
+  name     = "infra-ci-jenkins-io-controller" # Custom name on the secondary subscription (it is AKS managed on the primary)
+  location = var.location
+  tags     = local.default_tags
+}
+# Required to allow controller to check for subnets inside the sponsorship network
+resource "azurerm_role_definition" "infra_ci_jenkins_io_controller_vnet_sponsorship_reader" {
+  provider = azurerm.jenkins-sponsorship
+  name     = "Read-infra-ci-jenkins-io-sponsorship-VNET"
+  scope    = data.azurerm_virtual_network.private.id
```
Comment (dduportal) [inline infra.ci.jenkins.io.tf:103]: Should be:

```suggestion
scope = data.azurerm_virtual_network.infra_ci_jenkins_io_sponsorship.id
```

File: infra.ci.jenkins.io.tf [hunk 2]
```diff
@@ -67,3 +67,69 @@ resource "azurerm_role_assignment" "infra_ci_jenkins_io_privatek8s_subnet_privat
...
+
+  permissions {
+    actions = ["Microsoft.Network/virtualNetworks/read"]
+  }
+}
+resource "azurerm_role_assignment" "infra_controller_vnet_reader" {
+  provider           = azurerm.jenkins-sponsorship
+  scope              = data.azurerm_virtual_network.private.id
```
Comment (dduportal) [inline infra.ci.jenkins.io.tf:111]:
```suggestion
scope = data.azurerm_virtual_network.infra_ci_jenkins_io_sponsorship.id
```

File: infra.ci.jenkins.io.tf [hunk 3]
```diff
@@ -67,3 +67,69 @@ resource "azurerm_role_assignment" "infra_ci_jenkins_io_privatek8s_subnet_privat
...
+  }
+  source = "./.shared-tools/terraform/modules/azure-jenkinsinfra-azurevm-agents"
+
+  service_fqdn                     = local.infra_ci_jenkins_io_fqdn
+  service_short_stripped_name      = local.infra_ci_jenkins_io_service_short_stripped_name
+  ephemeral_agents_network_rg_name = data.azurerm_subnet.infra_ci_jenkins_io_sponsorship_ephemeral_agents.resource_group_name
+  ephemeral_agents_network_name    = data.azurerm_subnet.infra_ci_jenkins_io_sponsorship_ephemeral_agents.virtual_network_name
+  ephemeral_agents_subnet_name     = data.azurerm_subnet.infra_ci_jenkins_io_sponsorship_ephemeral_agents.name
+  controller_rg_name               = azurerm_resource_group.infra_ci_jenkins_io_controller_jenkins_sponsorship.name
+  controller_ips                   = data.azurerm_subnet.privatek8s_tier.address_prefixes # Pod IPs: controlelr IP may change in the pods IP subnet
```
Comment (dduportal) [inline infra.ci.jenkins.io.tf:127]: nit: typo in comment

**Pull request merged**